src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildBotData.java

```diff
@@ -76,6 +80,63 @@ public BuildBotData(String commitHash, String name, List<Log> logs, BuilderStatu
     this.status = status;
   }
 
+  public BuildBotData(@JsonProperty("builderName") String name) {
+    this.name = name;
+  }
+
+  /**
+   * Extracts nested commitHash("revision"), branch("branch") and
+   * timestamp("when") fields from the json
+   *
+   * @param sourceStamp Representation of the json component, where the commitHash field is located
+   */
+  @JsonProperty("sourceStamp")
+  @SuppressWarnings("unchecked")
+  public void unpackSourceStamp(Map<String, Object> sourceStamp) {
+    commitHash = sourceStamp.get("revision").toString();
+    List<Object> changes = (ArrayList<Object>)sourceStamp.get("changes");
+    Map<String, Object> latestChange = (Map<String, Object>)changes.get(0);
+    timestamp = Timestamp.ofTimeMicroseconds(Long.parseLong(latestChange.get("when").toString()));
+  }
+
+  /**
+   * Defines builder status based on a phrase provided in parsed json.
+   * If something failed or the phrase doesn't contain any of the key words
+   * ("failed", "successful", "lost"), then the buildbot is considered failed {@code FAILED}.
+   * If something is lost, then the buildbot is considered lost {@code LOST}.
+   * If  the buildbot is considered passed {@code PASSED}.
+   *
+   * @param words Words of the "text" JSON field
+   */
+  @JsonProperty("text")
+  private void extractStatus(List<String> words) {
+    for (String word : words) {
```

Comment:
Is it possible that the list is empty?

Reply:
I haven't seen that in real BuildBot API JSONs, but in this case the BuildBot will be considered as failed.